public/editor/scripts/project-list.js

```diff
@@ -30,16 +42,54 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
     return "{{ momentJSLastEdited | safe }}".replace("<% timeElapsed %>", timeElapsed);
   }
 
+  function updateFavorite(projectId, projectSelector, project){
+    var indexOfProjectInFavorites = favorites.indexOf(projectId);
+    var projectFavoriteButton = projectSelector + " .project-favorite-button";
+
+    if(indexOfProjectInFavorites !== -1){
+      favoriteProjectsElementList.push(project);
+      $(projectFavoriteButton).toggleClass("project-favorite-selected");
+      $(projectFavoriteButton).text("Unfavorite");
+    }
+
+    $(projectSelector + " .project-favorite").on("click", function() {
+      indexOfProjectInFavorites = favorites.indexOf(projectId);
+      projectFavoriteButton = projectSelector + " .project-favorite-button";
+
+      if(indexOfProjectInFavorites === -1) {
+        favorites.push(projectId);
+        localStorage.setItem("project-favorites", JSON.stringify(favorites));
+        $(projectFavoriteButton).toggleClass("project-favorite-selected");
+        $(projectFavoriteButton).text("Unfavorite");
+      }
+      else {
+        favorites.splice(indexOfProjectInFavorites, 1);
+        localStorage.setItem("project-favorites", JSON.stringify(favorites));
+        $(projectFavoriteButton).toggleClass("project-favorite-selected");
+        $(projectFavoriteButton).text("Favorite");
+      }
+    });
+  }
+
+  var favoriteProjectsElementList = [];
+
   Array.prototype.forEach.call(projects, function(project) {
     var projectSelector = "#" + project.getAttribute("id");
     var lastEdited = project.getAttribute("data-project-date_updated");
+    var projectId = project.getAttribute("data-project-id");
 
+    if(isLocalStorageAvailable) {
+      updateFavorite(projectId, projectSelector, project);
```

Comment:
Now that I think about it, can we rename this function to be `setFavoriteDataForProject`?